xbmc/epg/GUIEPGGridContainer.cpp

```diff
@@ -1429,9 +1429,11 @@ bool CGUIEPGGridContainer::OnMouseWheel(char wheel, const CPoint &point)
 
 CPVRChannelPtr CGUIEPGGridContainer::GetChannel(int iIndex)
 {
+  CSingleLock lock(m_critSection);
+
   if (iIndex >= 0 && (size_t) iIndex < m_channelItems.size())
   {
-    CFileItemPtr fileItem = m_channelItems[iIndex];
+    const CFileItemPtr &fileItem = m_channelItems[iIndex];
```

Comment:
Looks like you don't understand how reference counting works. If what you describe were possible, we would be doomed already, because taking the reference would already crash.
If you already own the reference, you don't need to take another one - and if you don't own the reference, it's impossible to create another one.